src/abstractdataframe/reshape.jl

```diff
@@ -32,9 +32,10 @@ that return views into the original data frame.
 - `view` : whether the stacked data frame should be a view rather than contain
   freshly allocated vectors.
 - `variable_eltype` : determines the element type of column `variable_name`.
-  By default a categorical vector of strings is created.
-  If `variable_eltype=Symbol` it is a vector of `Symbol`,
-  and if `variable_eltype=String` a vector of `String` is produced.
+  By default a `PooledArray{String}` is created.
+  If `variable_eltype=Symbol` it is a `PoolVector{Symbol}`,
+  and if `variable_eltype=CategoricalValue{String}`
```

Comment:
I guess we are more general here. `variable_eltype` can be any type as long as `copyto!` from `Vector{String}` will work. `CategoricalValue{String}` is just one particular case.

Reply:
I've improved the docstring. I've also changed the code a bit so that it produces a `PooledArray` unless the provided type returns a special array.

Reply:
maybe add to a docstring that this is for `view=false` as for `view=true` we have a `RepeatedVector`.

Reply:
Indeed.